packages/propeller/src/StreamsSink.mts

```diff
@@ -7,9 +7,132 @@ import EventEmitter from "node:events"
 
 const tracer = trace.getTracer("@equinox-js/propeller")
 
+namespace StreamSpan {
+  const idx = (span: ITimelineEvent[]) => span[0]!.index
+  const ver = (span: ITimelineEvent[]) => span.at(-1)!.index
+  export function merge(min, spans: ITimelineEvent[][]) {}
+}
+
+namespace Buffer {
+  const WritePosUnknown = -2n
```

Comment:
this stuff is encoded like this in F# to stop the need for a bool for malformed pushing the memory requirement from 16bits or something similar - consider using flags like the original code did rather than having such vestigial nonsense!